src/orange/OrangeData.hh

```diff
@@ -146,6 +149,41 @@ struct RaggedRightIndexerData
     }
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * Data for a single Bounding Inverval Hierarchy node
+ *
+ * Leaf nodes must contain exactly one bounding box
+ */
+struct BIHNode
+{
+    enum Edge : size_type
+    {
+        left = 0,
+        right = 1
+    };
+
+    // inner node only
+    Array<BIHNodeId, 2> children;
+    Array<float, 2> partitions;
+
+    // leaf only
+    ItemRange<LocalVolumeId> vol_ids;
```

Comment:
If these are going to have completely orthogonal data, maybe these should be two separate classes? If you can construct the tree so that the internal nodes are partitioned so that all leaves are afterward, you should be able to compactly store separate arrays with a `BIHNodeId` storing the index of the first leaf node (to get the `BIHNodeId`↔︎`BIHLeafId` mapping to index into the leaf array).